src/Spice86.Core/Emulator/InterruptHandlers/Dos/Ems/ExpandedMemoryManager.cs

```diff
@@ -0,0 +1,711 @@
+        }
+        if (EmmHandles.Count == EmmMemory.TotalPages) {
+            _state.AH = EmmStatus.EmmOutOfHandles;
+            return;
+        }
+        if (numberOfPagesToAlloc > EmmMemory.TotalPages) {
+            _state.AH = EmmStatus.NotEnoughEmmPages;
+            return;
+        }
+        _state.DX = AllocatePages(numberOfPagesToAlloc).HandleNumber;
+        _state.AH = EmmStatus.EmmNoError;
+    }
+
+    public EmmHandle AllocatePages(ushort numberOfPagesToAlloc, EmmHandle? newHandle = null) {
+        int key = newHandle?.HandleNumber ?? EmmHandles.Count;
+        newHandle ??= new() {
+            HandleNumber = (ushort)key
+        };
+        while (numberOfPagesToAlloc > 0) {
+            ushort pageNumber = (ushort)newHandle.PageMap.Count;
+            newHandle.PageMap.Add(pageNumber, new EmmPage(EmmPageSize){
+                PageNumber = pageNumber
+            });
+            numberOfPagesToAlloc--;
+        }
+
+        if (!EmmHandles.TryAdd(key, newHandle)) {
+            EmmHandles[key] = newHandle;
+        }
+        return newHandle;
+    }
+
+    /// <summary>
+    /// The Map/Unmap Handle Page function maps a logical page at a specific
+    /// physical page anywhere in the mappable regions of system memory. <br/>
+    /// The lowest valued physical page numbers are associated with regions of
+    /// memory outside the conventional memory range.  Use Function 25 (Get Mappable Physical Address Array)
+    /// to determine which physical pages within a system are mappable and determine the segment addresses which
+    /// correspond to a specific physical page number.  Function 25 provides a
+    /// cross reference between physical page numbers and segment addresses. <br/>
+    /// This function can also unmap physical pages, making them inaccessible for reading or writing. <br/>
+    /// You unmap a physical page by setting its associated logical page to FFFFh.
+    /// </summary>
+    /// <remarks>
+    /// The handle determines what type of pages are being mapped. <br/>
+    /// Logical pages allocated by Function 4 and Function 27 (Allocate Standard Pages
+    /// subFunction) are referred to as pages and are 16K bytes long. <br/>
+    /// Logical pages allocated by Function 27 (Allocate Raw Pages subFunction) are
+    /// referred to as raw pages and might not be the same size as logical pages.
+    /// </remarks>
+    public void MapUnmapHandlePage() {
+        ushort physicalPageNumber = _state.AL;
+        ushort logicalPageNumber = _state.BX;
+        ushort handleId = _state.DX;
+        _state.AH = MapUnmapHandlePage(logicalPageNumber, physicalPageNumber, handleId);
+    }
+
+    public byte MapUnmapHandlePage(ushort logicalPageNumber, ushort physicalPageNumber, ushort handleId) {
+        if (physicalPageNumber > EmmPageFrame.Count) {
+            if (_loggerService.IsEnabled(LogEventLevel.Debug)) {
+                _loggerService.Debug("Logical page {PhysicalPage} out of range",
```

Comment:
Does this happen a lot? Otherwise I'd think this could be an Information or even a Warning.

Reply:
loglevel set as Warning.